pkg/acquisition/modules/file/file.go

```diff
@@ -0,0 +1,386 @@
+			}
+		}
+		files, err := filepath.Glob(pattern)
+		if err != nil {
+			return errors.Wrap(err, "Glob failure")
+		}
+		if len(files) == 0 {
+			f.logger.Warnf("No matching files for pattern %s", pattern)
+			continue
+		}
+		f.logger.Infof("Will read %d files", len(files))
+		for _, file := range files {
+			if files[0] != pattern && f.config.Mode == configuration.TAIL_MODE { //we have a glob pattern
+				directory := path.Dir(file)
+				if !f.watchedDirectories[directory] {
+
+					err = f.watcher.Add(directory)
+					if err != nil {
+						f.logger.Errorf("Could not create watch on directory %s : %s", directory, err)
+						continue
+					}
+					f.watchedDirectories[directory] = true
+				}
+			}
+			f.logger.Infof("Adding file %s to filelist", file)
+			f.files = append(f.files, file)
+		}
+	}
+	return nil
+}
+
+func (f *FileSource) ConfigureByDSN(dsn string, labelType string, logger *log.Entry) error {
+	if !strings.HasPrefix(dsn, "file://") {
+		return fmt.Errorf("invalid DSN %s for file source, must start with file://", dsn)
+	}
+
+	pattern := strings.TrimPrefix(dsn, "file://")
+
+	if len(pattern) == 0 {
+		return fmt.Errorf("empty file:// DSN")
+	}
+
+	f.logger = logger
+	f.config = FileConfiguration{}
+	f.config.Labels = map[string]string{"type": labelType}
+	f.config.Mode = configuration.CAT_MODE
+
+	f.logger.Debugf("Will try pattern %s", pattern)
+	files, err := filepath.Glob(pattern)
+	if err != nil {
+		return errors.Wrap(err, "Glob failure")
+	}
+
+	if len(files) == 0 {
+		return fmt.Errorf("no matching files for pattern %s", pattern)
+	}
+
+	f.logger.Infof("Will read %d files", len(files))
+
+	for _, file := range files {
+		f.logger.Infof("Adding file %s to filelist", file)
+		f.files = append(f.files, file)
+	}
+	return nil
+}
+
+func (f *FileSource) GetMode() string {
+	return f.config.Mode
+}
+
+//SupportedModes returns the supported modes by the acquisition module
+func (f *FileSource) SupportedModes() []string {
+	return []string{configuration.TAIL_MODE, configuration.CAT_MODE}
+}
+
+//OneShotAcquisition reads a set of file and returns when done
+func (f *FileSource) OneShotAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	f.logger.Debug("In oneshot")
+	for _, file := range f.files {
+		fi, err := os.Stat(file)
+		if err != nil {
+			return fmt.Errorf("could not stat file %s : %w", file, err)
+		}
+		if fi.IsDir() {
+			f.logger.Warnf("%s is a directory, ignoring it.", file)
+			continue
+		}
+		f.logger.Infof("reading %s at once", file)
+		err = f.readFile(file, out, t)
+		if err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func (f *FileSource) GetMetrics() []prometheus.Collector {
+	return []prometheus.Collector{linesRead}
+}
+
+func (f *FileSource) GetName() string {
+	return "file"
+}
+
+func (f *FileSource) CanRun() error {
+	return nil
+}
+
+func (f *FileSource) StreamingAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	f.logger.Debug("Starting live acquisition")
```

Comment:
I think we want to customize logger here (with ie. pattern or unique identifier to be able to track)